kernel/dosfns.c

```diff
@@ -46,32 +46,48 @@ STATIC int remote_lock_unlock(sft FAR *sftp,    /* SFT for file */
                              unsigned long len, /* length (in bytes) of region to lock or unlock */
                              int unlock);       /* one to unlock; zero to lock */
 
-/* get current directory structure for drive
-   return NULL if the CDS is not valid or the
-   drive is not within range */
-struct cds FAR *get_cds(unsigned drive)
+static struct cds FAR *_get_cds(unsigned drive, int validate)
 {
   struct cds FAR *cdsp;
   unsigned flags;
 
   if (drive >= lastdrive)
     return NULL;
   cdsp = &CDSp[drive];
-  flags = cdsp->cdsFlags;
-  /* Entry is disabled or JOINed drives are accessable by the path only */
-  if (!(flags & CDSVALID) || (flags & CDSJOINED) != 0)
-    return NULL;
-  if (!(flags & CDSNETWDRV) && cdsp->cdsDpb == NULL)
-    return NULL;
+  if (validate) {
```

Comment:
Could you please just move this part to
`get_cds()` itself and avoid an extra parameter?

Reply:
The duplication is only `flags = CDSp[drive].flags`.
One line. If you don't want even a 1-line duplication,
you can pass these flags as a new parameter, so
that the main function doesn't have to retrieve them.
Passing various control-flow knobs to the function
is usually an anti-pattern.

Reply:
https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior

Reply:
Actually I coded it up with the duplication now and it looks better. Thanks for the link, interesting.